### ISSUE TITLE
("pack", UrwidImage(...))  causes the UrwidImage to not render

### ISSUE THREAD

**Issue by danschwarz** [AnonymouX47/term-image#77]
I have an urwid.Columns widget in which I am placing text in some columns, an UrwidImage in another. ("pack", urwid.Text("SomeTextHere"))  works fine.  ("pack",BoxAdapter(UrwidImage(...)),1) causes the UrwidImage to disappear on render.  I think it may be calculating zero-width.  Thoughts?

**Comment by AnonymouX47** [comment]
Yes, `UrwidImage` isn't a [fixed widget](http://urwid.org/manual/widgets.html#box-flow-and-fixed-widgets).

The reason being, I could think of any way to compute a valid image size without *maxcols* or *maxrows* i.e an empty *size* tuple, and to be fair, it doesn't make much sense anyways.

Use `Columns([..., UrwidImage(...), ...])` instead i.e as a box/flow widget with weighted size within the `Columns` widget.

**Comment by AnonymouX47** [comment]
With `Columns(<list of `UrwidImage` widgets>)`:

![Screenshot_2023-02-27_00-37-58](https://user-images.githubusercontent.com/61663146/221444437-fb9f90ba-46c9-4979-9b6f-920e65c80613.png)

**Comment by AnonymouX47** [comment]
This is totally intended behavior. See:

https://github.com/AnonymouX47/term-image/blob/10ba5add52d33f4f21d9462804b4a2cb9333cbd1/src/term_image/widget/urwid.py#L160

Though, I guess i need to change "packed" to "fixed". That was a mistake.

**Comment by AnonymouX47** [comment]
I'm really sorry, I should've asked first. Any ideas on how to compute the size for a fixed image widget?

**Comment by danschwarz** [comment]
>Only the first widget is rendered and fills the entire Columns widget. So, I guess that's what you're experiencing.

That is what I'm experiencing. After I wrote up the issue, I realized that it wasn't a zero width situation, but a too-wide situation preventing the display of the image.  

I'm able to work around this issue by simply not attempting to pack the UrwidImage. As long as I follow it with another column of sufficient weight, the UrwidImage gets displayed at minimum width, which is what I am after.

>Any ideas on how to compute the size for a fixed image widget?
Given that you do centering and (optionally) upscaling, I don't think it is possible to know the size outside of render()
